include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -2366,122 +2343,9 @@ __parallel_find_or_impl(oneapi::dpl::__internal::__device_backend_tag, _Executio
-{
-    auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator1>();
-    auto __buf = __keep(__first, __last);
-    auto __s_keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator2>();
-    auto __s_buf = __s_keep(__s_first, __s_last);
-
-    return oneapi::dpl::__par_backend_hetero::__parallel_find_or(
-        oneapi::dpl::__internal::__device_backend_tag{},
-        __par_backend_hetero::make_wrapped_policy<__or_policy_wrapper>(::std::forward<_ExecutionPolicy>(__exec)), __f,
-        __parallel_or_tag{}, __buf.all_view(), __s_buf.all_view());
-}
-
-// Special overload for single sequence cases.
-// TODO: check if similar pattern may apply to other algorithms. If so, these overloads should be moved out of
-// backend code.
-template <typename _ExecutionPolicy, typename _Iterator, typename _Brick>
-bool
-__parallel_or(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Iterator __first,
-              _Iterator __last, _Brick __f)
-{
-    auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator>();
-    auto __buf = __keep(__first, __last);
-
-    return oneapi::dpl::__par_backend_hetero::__parallel_find_or(
-        oneapi::dpl::__internal::__device_backend_tag{},
-        __par_backend_hetero::make_wrapped_policy<__or_policy_wrapper>(::std::forward<_ExecutionPolicy>(__exec)), __f,
-        __parallel_or_tag{}, __buf.all_view());
-}
-
-//------------------------------------------------------------------------
-// parallel_find - sync pattern
-//-----------------------------------------------------------------------
-
-template <typename Name>
-class __find_policy_wrapper
-{
-};
-
-template <typename _ExecutionPolicy, typename _Iterator1, typename _Iterator2, typename _Brick, typename _IsFirst>
-_Iterator1
-__parallel_find(oneapi::dpl::__internal::__device_backend_tag, _ExecutionPolicy&& __exec, _Iterator1 __first,
-                _Iterator1 __last, _Iterator2 __s_first, _Iterator2 __s_last, _Brick __f, _IsFirst)
-{
-    auto __keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator1>();
-    auto __buf = __keep(__first, __last);
-    auto __s_keep = oneapi::dpl::__ranges::__get_sycl_range<__par_backend_hetero::access_mode::read, _Iterator2>();
-    auto __s_buf = __s_keep(__s_first, __s_last);
-
-    using _TagType = ::std::conditional_t<_IsFirst::value, __parallel_find_forward_tag<decltype(__buf.all_view())>,
-                                          __parallel_find_backward_tag<decltype(__buf.all_view())>>;
-    return __first + oneapi::dpl::__par_backend_hetero::__parallel_find_or(
-                         oneapi::dpl::__internal::__device_backend_tag{},
-                         __par_backend_hetero::make_wrapped_policy<__find_policy_wrapper>(
-                             ::std::forward<_ExecutionPolicy>(__exec)),
```

Comment:
I'm struggling a bit to understand why we needed this wrapped policy layer previously (and if we would need something similar now).  I don't think it should be required here or in the new code because we are not calling multiple kernels from the same public API call (which is where we should need such wrapped policies).  
Please comment if you agree with this assessment or if there is something else here I'm missing.

Reply:
In my understanding, in the past, we thought that we have to provide one unique kernel name by each "dispatching patch" of any algorithm call.
We have many cases when several algorithm is dispatching to one middle layer pattern.
So, on each step of dispatching a device policy was wrapped into new one. It  made more or less guarantee that on final SYCL pattern we got (from the  policy) more or less unique kernel name...

But, now, I think such approach is not necessary. SYCL compiler can share kernels if we don't use lambdas..